Packs/CommonScripts/Scripts/PFXAnalyzer/PFXAnalyzer.py

```diff
@@ -0,0 +1,240 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+from cryptography.hazmat.primitives.serialization import pkcs12
+from cryptography import x509
+from cryptography.x509.oid import NameOID, ExtensionOID
+from cryptography.hazmat.primitives.asymmetric import rsa, ec
+from cryptography.exceptions import InvalidSignature
+from cryptography.x509.extensions import CRLDistributionPoints, AuthorityInformationAccess
+import os
+from datetime import datetime
+
+
+SUSPICIOUS_KEYWORDS = [
+    keyword.lower()
+    for keyword in [
+        "Microsoft",
+        "Google",
+        "Apple",
+        "Adobe",
+        "Facebook",
+        "Amazon",
+        "NVIDIA",
+        "Cisco",
+        "Intel",
+        "Oracle",
+        "Symantec",
+    ]
+]
+
+
+def get_cn(cert_name_object):
+    for attr in cert_name_object:
+        if attr.oid == NameOID.COMMON_NAME:
+            return attr.value
+    return ""
+
+
+def is_self_signed(cert):
+    try:
+        cert.public_key().verify(
+            cert.signature,
+            cert.tbs_certificate_bytes,
+            cert.signature_algorithm_oid._name,
+            cert.signature_hash_algorithm,
+        )
+        return cert.issuer == cert.subject
+    except InvalidSignature:
+        return False
+    except Exception:
+        return False
+
+
+def analyze_pfx_file(pfx_data: bytes, pfx_password: Optional[str] = None) -> dict:
```

Comment:
@karinafishman In addition to Dan's comments, to improve readability and maintainability, let's refactor analyze_pfx_file by extracting larger, distinct logic blocks into their own helper functions